File: 2017-04-11/ikselven_euler87.php
#!/usr/bin/php
<?php
define("MAX", 5*(10**7));

$nextPrime = 2;
$primes = array($nextPrime);
while ($nextPrime < sqrt(MAX)) {
    $nextPrime = gmp_intval(gmp_nextprime($nextPrime));
    $primes[] = $nextPrime;
}

$sums = array();
$primeCount = count($primes);
for($outer = 0; $outer < $primeCount; $outer++) {
    $fourth = $primes[$outer] ** 4;
    if ($fourth >= MAX) {
        break;
    }
    
    for($middle = 0; $middle < $primeCount; $middle++) {
        $cube = $fourth + $primes[$middle] ** 3;
        if ($cube >= MAX) {
            break;
        }

        for($inner = 0; $inner < $primeCount; $inner++) {
            $sum = $cube + $primes[$inner] ** 2;
            if ($sum >= MAX) {
                break;
            }
            $sums[] = $sum;
        }
    }
}

echo "Found ".count(array_unique($sums))." numbers\n";

File: 2017-04-11/ikselven_euler87.R
library(numbers)
MAX <- 5e7
primes <- Primes(1, ceiling(sqrt(MAX)))

potfilter <- function(numbers, filter)
    numbers[numbers < filter]

sums <- sapply(
    potfilter(primes^4, MAX),
    function(x) { 
        sapply(
            potfilter(primes^3, MAX),
            function(y) { 
                sapply(primes^2, function(z) {
                    x+y+z
                })
            }
        )
    }
)

length(unique(sums[sums<MAX]))
